Installation issue: memkind

### Steps to reproduce the issue

```console
$ spack spec -I memkind
Input spec
--------------------------------
 -   memkind

Concretized
--------------------------------
 -   memkind@1.13.0%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]      ^autoconf@2.69%gcc@13.0.1 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-fedora38-zen2
[+]          ^perl@5.36.0%gcc@13.0.1+cpanm+open+shared+threads build_system=generic arch=linux-fedora38-zen2
[+]              ^berkeley-db@18.1.40%gcc@13.0.1+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-fedora38-zen2
[+]              ^bzip2@1.0.8%gcc@13.0.1~debug~pic+shared build_system=generic arch=linux-fedora38-zen2
[+]              ^gdbm@1.23%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]                  ^readline@8.2%gcc@13.0.1 build_system=autotools patches=bbf97f1 arch=linux-fedora38-zen2
[+]                      ^ncurses@6.4%gcc@13.0.1~symlinks+termlib abi=none build_system=autotools arch=linux-fedora38-zen2
[+]                          ^pkgconf@1.8.0%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]              ^zlib@1.2.13%gcc@13.0.1+optimize+pic+shared build_system=makefile arch=linux-fedora38-zen2
[+]      ^automake@1.16.5%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]      ^libtool@2.4.7%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]      ^m4@1.4.19%gcc@13.0.1+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-fedora38-zen2
[+]          ^diffutils@3.9%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]              ^libiconv@1.17%gcc@13.0.1 build_system=autotools libs=shared,static arch=linux-fedora38-zen2
[+]          ^libsigsegv@2.14%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]      ^numactl@2.0.14%gcc@13.0.1 build_system=autotools patches=4e1d78c,62fc8a8,ff37630 arch=linux-fedora38-zen2
```


### Error message

<details><summary>Error message</summary>

<pre>
==> Installing memkind-1.13.0-oamsbvioutdvby42kkunto46vwxyrnro
==> No binary for memkind-1.13.0-oamsbvioutdvby42kkunto46vwxyrnro found: installing from source
==> Using cached archive: /opt/spack/var/spack/cache/_source-cache/archive/3f/3f0d919b61fdd4d2ebce14e0b7dbb856e2144138778940107c13549523f3bdc0.tar.gz
==> Ran patch() for memkind
==> memkind: Executing phase: 'autoreconf'
==> memkind: Executing phase: 'configure'
==> memkind: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'V=1'

7 errors found in build log:
     5      libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
     6      libtoolize: copying file 'm4/libtool.m4'
     7      libtoolize: copying file 'm4/ltoptions.m4'
     8      libtoolize: copying file 'm4/ltsugar.m4'
     9      libtoolize: copying file 'm4/ltversion.m4'
     10     libtoolize: copying file 'm4/lt~obsolete.m4'
  >> 11     configure.ac:18: installing './ar-lib'
  >> 12     configure.ac:18: installing './compile'
  >> 13     configure.ac:23: installing './config.guess'
  >> 14     configure.ac:23: installing './config.sub'
  >> 15     configure.ac:14: installing './install-sh'
  >> 16     configure.ac:14: installing './missing'
     17     Makefile.am: installing './depcomp'
     18     parallel-tests: installing './test-driver'
     19     ==> memkind: Executing phase: 'configure'
     20     ==> [2023-04-28-19:01:30.785914] FILTER FILE: /tmp/root/spack-stage/spack-stage-memkind-1.13.0-oamsbvioutdvby42kkunto46vw
            xyrnro/spack-src/configure [replacing "^(\s*if test x-L = )("\$p" \|\|\s*)$"]
     21     ==> [2023-04-28-19:01:30.804116] FILTER FILE: /tmp/root/spack-stage/spack-stage-memkind-1.13.0-oamsbvioutdvby42kkunto46vw
            xyrnro/spack-src/configure [replacing "^(\s*test x-R = )("\$p")(; then\s*)$"]
     22     ==> [2023-04-28-19:01:30.819797] FILTER FILE: /tmp/root/spack-stage/spack-stage-memkind-1.13.0-oamsbvioutdvby42kkunto46vw
            xyrnro/spack-src/configure [replacing "^(\s*test \$p = "-R")(; then\s*)$"]

     ...

     171    ==> [2023-04-28-19:01:34.110380] FILTER FILE: /tmp/root/spack-stage/spack-stage-memkind-1.13.0-oamsbvioutdvby42kkunto46vw
            xyrnro/spack-src/libtool [replacing "^pic_flag=""$"]
     172    ==> [2023-04-28-19:01:34.149014] FILTER FILE: /tmp/root/spack-stage/spack-stage-memkind-1.13.0-oamsbvioutdvby42kkunto46vw
            xyrnro/spack-src/libtool [replacing "^pic_flag=""$"]
     173    ==> [2023-04-28-19:01:34.186604] FILTER FILE: /tmp/root/spack-stage/spack-stage-memkind-1.13.0-oamsbvioutdvby42kkunto46vw
            xyrnro/spack-src/libtool [replacing "^pic_flag=""$"]
     174    ==> memkind: Executing phase: 'build'
     175    ==> [2023-04-28-19:01:34.244290] 'make' '-j16' 'V=1'
     176    [ ! -e jemalloc/configure ] && (cd jemalloc && autoconf) || exit 0
  >> 177    make: *** No rule to make target 'jemalloc/lib/libjemalloc_pic.a', needed by 'libmemkind.la'.  Stop.
     178    make: *** Waiting for unfinished jobs....
     179    [ ! -e jemalloc/lib/libjemalloc_pic.a ] && (cd jemalloc && ./configure --enable-autogen  --without-export --with-version=
            5.2.1-0-gea6b3e973b477b8061e0076bb257dbd7f3faa756 --disable-fill  --disable-initial-exec-tls --with-jemalloc-prefix=jemk_
             --with-malloc-conf="narenas:256,lg_tcache_max:12"  && make) || exit 0
     180    checking for xsltproc... false
     181    checking for gcc... /opt/spack/lib/spack/env/gcc/gcc
     182    checking whether the C compiler works... yes
     183    checking for C compiler default output file name... a.out

See build log for details:
  /tmp/root/spack-stage/spack-stage-memkind-1.13.0-oamsbvioutdvby42kkunto46vwxyrnro/spack-build-out.txt

</pre></details>


### Information on your system

* **Spack:** 0.20.0.dev0 (4363b1c2dcdae9245054cd44124f8ac6789bfd39)
* **Python:** 3.11.3
* **Platform:** linux-fedora38-zen2
* **Concretizer:** clingo

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/11356566/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/11356567/spack-build-env.txt)

Note that installing with `-j1` does work, so it seems to be an issue related to the late `make` 4.4 in Fedora 38.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/memkind/package.py
+++ b/var/spack/repos/builtin/packages/memkind/package.py
@@ -42,6 +42,9 @@ class Memkind(AutotoolsPackage):
     # <https://github.com/memkind/memkind#jemalloc>.
     conflicts("jemalloc")
 
+    # https://github.com/spack/spack/issues/37292
+    parallel = False
+
     def patch(self):
         with open("VERSION", "w") as version_file:
             version_file.write("{0}\n".format(self.version))

